rclcpp/src/rclcpp/duration.cpp

```diff
@@ -246,6 +246,26 @@ Duration::to_rmw_time() const
   return result;
 }
 
+Duration
+Duration::from_rmw_time(rmw_time_t duration)
+{
+  Duration ret;
+  constexpr rcl_duration_value_t limit = std::numeric_limits<rcl_duration_value_t>::max();
+  if (duration.sec > limit || duration.nsec > limit) {
```

Comment:
Should it be something like:

```suggestion
  if (RCL_S_TO_NS(duration.sec) > limit || duration.nsec > limit) {
```

Just because the value of `.sec` could be much smaller than `limit` but still overflow `ret.rcl_duration_.nanoseconds`.

Reply:
thanks, solved in https://github.com/ros2/rclcpp/pull/1467/commits/655f76e14c95876951e268c7c3d63bc185ede1a1